framework/src/transfers/MultiAppGeneralFieldTransfer.C

```diff
@@ -848,8 +873,18 @@ MultiAppGeneralFieldTransfer::examineLocalValueConflicts(
       auto & fe_type = to_sys->variable_type(var_num);
       bool is_nodal = _to_variables[var_index]->isNodal();
 
-      // Move to target problem local mesh
-      auto local_p = p + _to_positions[i_to];
+      // Move to the local frame of reference for the target
+      Point local_p;
+      const auto & to_transform = _to_transforms[getGlobalTargetAppIndex(i_to)];
+      if (to_transform->hasNonTranslationTransformation())
+      {
+        if (!_skip_coordinate_collapsing)
+          mooseInfo("Resolution of local value conflicts detected cannot find the exact origin "
+                    "point due to the non-uniqueness of the coordinate collapsing reverse mapping");
+        local_p = p - _to_positions[i_to];
+      }
+      else
+        local_p = to_transform->mapBack(p);
```

Comment:
still don't get it haha. Also if this is going to be a repeated pattern, is it possible to create a method and use it instead of repeating?

Reply:
added new APIs